src/gui/spatialArrow.js

```diff
@@ -0,0 +1,324 @@
+    const menuBarHeight = 30;
+
+    function addCanvas() {
+        canvasContainer = document.createElement('div');
+        canvasContainer.className = 'arrow-canvas-container';
+        canvasContainer.style.position = 'absolute';
+        canvasContainer.style.top = '0';
+        canvasContainer.style.left = '0';
+        document.body.appendChild(canvasContainer);
+
+        canvas = document.createElement('canvas');
+        canvas.className = 'arrow-canvas';
+        canvas.style.position = 'absolute';
+        canvas.style.top = `${menuBarHeight}px`;
+        canvas.style.left = '0';
+        canvas.style.zIndex = '3001';
+        canvas.style.border = '2px solid black';
+        canvasContainer.appendChild(canvas);
+
+        ctx = canvas.getContext("2d");
+    }
+
+    function resizeCanvas() {
+        if (canvas !== undefined) {
+            screenW = window.innerWidth * windowRatio;
+            screenH = (window.innerHeight - menuBarHeight) * windowRatio;
+            screenRatio = screenH / screenW;
+            canvas.width = screenW;
+            canvas.height = screenH;
+        }
+    }
+
+    function Rot(a) {
+        let s = sin(a), c = cos(a);
+        return [[c, -s], [s, c]];
+    }
+
+    const Clamp = (x, low, high) => {
+        return Math.min(Math.max(x, low), high);
+    }
+
+    const Remap01 = (x, low, high) => {
+        return Clamp((x - low) / (high - low), 0, 1);
+    }
+
+    const Remap = (x, low1, high1, low2, high2) => {
+        return low2 + (high2 - low2) * Remap01(x, low1, high1);
+    }
+
+    let translateX = 0, translateY = 0;
+
+    function Translate(x, y) {
+        translateX += x;
+        translateY += y;
+        ctx.translate(x, y);
+    }
+
+    let rotation = 0;
+
+    function Rotate(a) {
+        rotation += a;
+        ctx.rotate(a);
+    }
+
+    let scaleX = 1, scaleY = 1;
+
+    function Scale(x, y) {
+        scaleX *= x;
+        scaleY *= y;
+        ctx.scale(x, y);
+    }
+
+    function clear() {
+        ctx.clearRect(-translateX, -translateY, screenW, screenH);
+    }
+
+    // draw a rect at (x, y) center
+    function Rect(x, y, width, height, rotation = 0) {
+        Translate(x, y);
+        Rotate(rotation);
+        ctx.fillRect(-width / 2,-height / 2, width, height);
+        Rotate(-rotation);
+        Translate(-x, -y);
+    }
+
+    // draw an arrow at (x, y) center
+    function Arrow(x, y, rotation, scale, innerColor='rgb(0, 255, 255)', outerColor='rgb(255, 255, 255)') {
+        Translate(x, y);
+        Scale(scale, scale);
+        Rotate(rotation);
+        
+        // draw path
+        let region = new Path2D();
+        region.moveTo(0, -3);
+        region.lineTo(-2, 2);
+        // region.lineTo(0, 1);
+        region.lineTo(2, 2);
+        region.closePath();
+        // fill path
+        ctx.fillStyle = innerColor;
+        // ctx.fillStyle = innerColor;
```

Comment:
If you remove some of the commented-out code, that will make it more easily readable